fastify.d.ts

```diff
@@ -6,18 +6,23 @@ import * as http from 'http';
 import * as http2 from 'http2';
 import * as https from 'https';
 import * as pino from 'pino';
+import * as express from 'express';
 
-declare function fastify<HttpServer, HttpRequest, HttpResponse>(opts?: fastify.ServerOptions): fastify.FastifyInstance<HttpServer, HttpRequest, HttpResponse>;
+declare function fastify<
+  HttpServer extends http.Server = http.Server, 
+  HttpRequest extends http.IncomingMessage = http.IncomingMessage, 
+  HttpResponse extends http.ServerResponse = http.ServerResponse
+>(opts?: fastify.ServerOptions): fastify.FastifyInstance<HttpServer, HttpRequest, HttpResponse>;
```

Comment:
We also support Http2. Not sure how to specify this here.

Reply:
This is just defaulting the case where you did not pass any generic type.
http2 support is enabled below based on wether or not the `http2` option was set to `true`

I dont think it should affect http2 let me know if you have a test case which its breaking and I'll work on fixing it,

Reply:
I'm grand with how it is. I do not know how to fix it - I wrote my first piece of TS yesterday and it was __very hard__.

Reply:
You were having trouble with TS and fastify? Or in general?

Reply:
In general, I even fixed a bug :D. https://github.com/fastify/fastify/pull/736

Reply:
cool 👍 

I've been using TS for a little while, but am now trying to see what it would take for use to use fastify for a new project.

Thats what driving me to fix up the types 😄

Reply:
@tikotzky these will break http2 type support: `HttpServer extends http.Server`. We support passing in the generics so correct typings can be determined later in handlers etc. and this includes http2 typings. http2 doesn't use the same structure as http so we can't enforce that extension.

Reply:
Perhaps a union type can solve this? For example:

```
declare function fastify<
   HttpServer extends http.Server | HttpServer extends http2.Server
   // etc
 >
```

Although it seems like http2 does not export the server class - https://github.com/nodejs/node/blob/master/lib/http2.js

Reply:
syntax for a union type would be
```js
declare function fastify<
  HttpServer extends (http.Server | http2.Http2Server) = http.Server, 
  HttpRequest extends (http.IncomingMessage | http2.Http2ServerRequest) = http.IncomingMessage, 
  HttpResponse extends (http.ServerResponse | http2.Http2ServerResponse) = http.ServerResponse
>...
```

and it does seem to work.
Gonna push that and a test for it too...